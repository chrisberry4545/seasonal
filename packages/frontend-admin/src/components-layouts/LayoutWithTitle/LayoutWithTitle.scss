@import '../../styles/settings';

.c-form-layout__title {
  border-bottom: 1px solid $grey-med;
  display: block;
  margin-bottom: 1.75rem;
  padding-bottom: 0.5rem;
}
