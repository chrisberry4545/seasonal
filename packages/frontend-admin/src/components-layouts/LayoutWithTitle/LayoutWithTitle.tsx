import React, { FC } from 'react';
import './LayoutWithTitle.scss';
import { TextHeadingMedium } from '@chrisb-dev/seasonal-shared-frontend-components';

interface ILayoutWithTitle {
  title?: string;
}

export const LayoutWithTitle: FC<ILayoutWithTitle> = ({
  children,
  title
}) => (
  <div className='c-layout-with-title'>
    {
      title
      && <TextHeadingMedium className='c-layout-with-title__title'>
        {title}
      </TextHeadingMedium>
    }
    { children }
  </div>
);
