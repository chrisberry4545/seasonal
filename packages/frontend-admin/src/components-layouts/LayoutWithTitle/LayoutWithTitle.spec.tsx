import React from 'react';
import { LayoutWithTitle } from './LayoutWithTitle';
import { shallow, ShallowWrapper } from 'enzyme';
import { TextHeadingMedium } from '@chrisb-dev/seasonal-shared-frontend-components';

jest.mock('@chrisb-dev/seasonal-shared-frontend-components', () => ({
  TextHeadingMedium: () => 'TextHeadingMedium'
}));

describe('<LayoutWithTitle />', () => {
  let wrapper: ShallowWrapper;

  describe('when there is a title set', () => {
    beforeEach(() =>
      wrapper = shallow(
        <LayoutWithTitle title='Title'>
          Children
        </LayoutWithTitle>
      )
    );

    test('renders correctly', () => expect(wrapper).toMatchSnapshot());

  });

  describe('when there is no title set', () => {
    beforeEach(() =>
      wrapper = shallow(
        <LayoutWithTitle>
          Children
        </LayoutWithTitle>
      )
    );

    test('does not show a title', () =>
      expect(wrapper.find(TextHeadingMedium).exists()).toBe(false));

  });

});
