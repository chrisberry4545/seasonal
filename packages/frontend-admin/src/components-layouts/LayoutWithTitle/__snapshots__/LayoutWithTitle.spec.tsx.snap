// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LayoutWithTitle /> when there is a title set renders correctly 1`] = `
<div
  className="c-layout-with-title"
>
  <TextHeadingMedium
    className="c-layout-with-title__title"
  >
    Title
  </TextHeadingMedium>
  Children
</div>
`;
